dashboard/app/views/levels/editors/_dsl.html.haml

```diff
@@ -9,6 +9,35 @@
 
 - if [ContractMatch, Match, Multi, TextMatch, External].include? @level.class
   %p Markdown Editor
-  #epiceditor{style: 'height: 350px; border: 5px;'}
-  = stylesheet_link_tag    'editor/markdown_editor', media: 'all'
-  = javascript_include_tag 'editor/markdown_editor'
+
+  %textarea#level_markdown_textarea
+
+  #markdown-preview
+
+:javascript
+  var regex = /markdown <<(\w*)\n([\s\S]*)\n\1$/m;
+  var dslElement = $('#level_dsl_text');
+  var dslText = dslElement.val();
+
+  var mdEditor = codeMirror('markdown_textarea', 'markdown', function (editor, change) {
+    $('#markdown-preview').html(marked(editor.getValue()));
+
+    var editorText = editor.getValue();
+    var dslText = dslElement.val();
+    var replacedText;
+    if (regex.exec(dslText)) {
+      replacedText = dslText.replace(regex,'markdown <<$1\n' + editorText + '\n$1');
+    } else {
+      replacedText = dslText + '\nmarkdown <<MARKDOWN\n' + editorText + '\nMARKDOWN';
+    }
+    dslElement.val(replacedText);
+  }, true);
+
+  // Match against markdown heredoc syntax and capture contents in [2].
+  var match = regex.exec(dslText);
+  if (match && match[2]) {
+    var markdownText = match[2];
+    mdEditor.setValue(markdownText);
+  } else {
+    mdEditor.setValue('');
+  }
```

Comment:
Ah, I can extract it into a js file if that'd be cleaner. I moved it over because the code was specifically referencing the `level_dsl_text` and I assumed the `editor/markdown_editor` JS and CSS were mostly intended to do the inclusion of epiceditor